tensorflow/python/training/checkpointable/util_test.py

```diff
@@ -1324,6 +1324,21 @@ def test_initialize_if_not_restoring(self):
       train_fn()
       self.assertEqual(42., self.evaluate(optimizer.variables()[0]))
 
+  @test_util.run_in_graph_and_eager_modes
+  def test_restore_after_adding_empty_checkpointable_data_structure(self):
+    with ops.Graph().as_default(), self.session(graph=ops.get_default_graph()):
+      model = NonLayerCheckpointable()
+      checkpoint = checkpointable_utils.Checkpoint(model=model)
+      checkpoint.restore(None).initialize_or_restore()
+      checkpoint_directory = self.get_temp_dir()
+      checkpoint_prefix = os.path.join(checkpoint_directory, "ckpt")
+      save_path = checkpoint.save(checkpoint_prefix)
+    with ops.Graph().as_default(), self.session(graph=ops.get_default_graph()):
+      model = NonLayerCheckpointable()
+      model.dict = {"a": 1}
+      model.list = {"b": 1}
+      checkpoint.restore(save_path).assert_consumed().run_restore_ops()
```

Comment:
Swap `assert_consumed` for `assert_existing_objects_matched`? `assert_consumed` is trickier and not the assertion your PR is modifying (so presumably this test will fail as it is now).

Reply:
Done and done :)